app/models/carto/visualization.rb

```
@@ -96,6 +96,10 @@ class Carto::Visualization < ActiveRecord::Base
   after_save :save_named_map_or_rollback_privacy, :propagate_attribution_change
   after_save :propagate_privacy_and_name_to, if: :table
 
+  before_destroy :backup_visualization
```

I think that we should follow the same approach than we did in the previous PR, and log usages. It was really useful. What do you think?

PS: for acceptance, I'd raise instead of logging in those cases that are not legit.